Classes/BagBuilder/Abstract.php

```diff
@@ -169,6 +169,49 @@ public function getWhereClause()
         return implode(' AND ', $this->whereClauseParts);
     }
 
+    /**
+     * Returns a WHERE clause part for the bag to create.
+     *
+     * If the bag does not have such limitation imposed upon, the return value
+     * will be empty.
+     *
+     * @param string $key the limitation key to return, must not be empty
+     *
+     * @return string WHERE clause part for the bag to create, may be empty
+     */
+    public function getWhereClausePart($key)
+    {
+        if (empty($key)) {
+            throw new \InvalidArgumentException('The parameter $key must not be empty.', 1569331331);
+        }
+
+        if (empty($this->whereClauseParts[$key])) {
+            return '';
+        }
+
+        return $this->whereClauseParts[$key];
```

Comment:
I think you can use the null coalesce operator to combine this line and the `if` statement before it.